src/windows/ride.c

```diff
@@ -2449,51 +2450,19 @@ static void window_ride_vehicle_mousedown(int widgetIndex, rct_window *w, rct_wi
 
 		dropdown_set_checked(selectedIndex, true);
 		break;
-	case WIDX_VEHICLE_TRAINS_DROPDOWN:
-		window_dropdown_show_text_custom_width(
-			w->x + dropdownWidget->left,
-			w->y + dropdownWidget->top,
-			dropdownWidget->bottom - dropdownWidget->top + 1,
-			w->colours[1],
-			DROPDOWN_FLAG_STAY_OPEN,
-			ride->max_trains,
-			widget->right - dropdownWidget->left
-		);
-
-		stringId = RideNameConvention[ride->type].vehicle_name + 4;
-		for (i = 0; i < 32; i++) {
-			gDropdownItemsFormat[i] = 1142;
-			gDropdownItemsArgs[i] = ((i + 1) << 16) | (i == 0 ? stringId : stringId + 1);
-		}
-
-		dropdown_set_checked(ride->num_vehicles - 1, true);
+	case WIDX_VEHICLE_TRAINS_INCREASE:
+			if(ride->num_vehicles<32)ride_set_num_vehicles(w->number, ride->num_vehicles + 1);
 		break;
-	case WIDX_VEHICLE_CARS_PER_TRAIN_DROPDOWN:
-		minCars = (ride->min_max_cars_per_train >> 4);
-		maxCars = (ride->min_max_cars_per_train & 0x0F);
-
-		window_dropdown_show_text_custom_width(
-			w->x + dropdownWidget->left,
-			w->y + dropdownWidget->top,
-			dropdownWidget->bottom - dropdownWidget->top + 1,
-			w->colours[1],
-			DROPDOWN_FLAG_STAY_OPEN,
-			maxCars - minCars + 1,
-			widget->right - dropdownWidget->left
-		);
-
-		for (i = 0; i < 12; i++) {
-			cars = minCars + i;
-
-			gDropdownItemsFormat[i] = 1142;
-			gDropdownItemsArgs[i] = 1024;
-			if (cars - rideEntry->zero_cars > 1)
-				gDropdownItemsArgs[i]++;
-			gDropdownItemsArgs[i] |= (cars - rideEntry->zero_cars) << 16;
-		}
-
-		dropdown_set_checked(ride->num_cars_per_train - minCars, true);
+	case WIDX_VEHICLE_TRAINS_DECREASE:
+			if(ride->num_vehicles>1)ride_set_num_vehicles(w->number, ride->num_vehicles - 1);
```

Comment:
Spacing here needs adjusting as well. The if is also intended a tab too far.
Both things apply to all three ifs.